rest-service/manager_rest/server.py

```diff
@@ -182,10 +186,77 @@ def create_logger(logger_name,
 
 
 def init_secured_app(_app):
+
+    def register_auth_token_generator(auth_token_generator):
+        _app.logger.debug('registering auth token generator {0}'
+                          .format(auth_token_generator))
+        _app.auth_token_generator = create_instance(auth_token_generator)
+
+    def register_userstore_driver(userstore_driver):
+        secure_app.app.logger.debug('registering userstore driver {0}'
+                                    .format(userstore_driver))
+        secure_app.userstore_driver = create_instance(userstore_driver)
+
+    def register_authentication_providers(authentication_providers):
+        # Note: the order of registration is important here
+        for provider in authentication_providers:
+            secure_app.app.logger.debug(
+                'registering authentication provider {0}'.format(provider))
+            secure_app.register_authentication_provider(
+                provider['name'], create_instance(provider))
+
+    def register_authorization_provider(authorization_provider):
+        secure_app.app.logger.debug('registering authorization provider {0}'
+                                    .format(authorization_provider))
+        # validate the roles_config.yaml file exists
+        roles_config_file_path = \
```

Comment:
True, and we need to discuss that. I just left it like this because of issues with a better implementation I am working on.